oxidation/libparsec/crates/types/src/regex.rs

```diff
@@ -11,14 +12,28 @@ fn escape_globing_pattern(string: &str) -> String {
     str::replace(string, "$", "\\$")
 }
 
+/// This function remove the `\Z` and `\\` sequences that are unsupported by the
+/// rust's regex. These escapes can still be in the regex strings if the user
+/// has a cached regex string generated by the python module
+fn remove_unsupported_escapes(string: &str) -> String {
+    string.replace("\\Z", "\\z")
+}
+
 impl Regex {
     pub fn from_pattern(pattern: &str) -> Result<Self, fnmatch_regex::error::Error> {
         let escaped_str = escape_globing_pattern(pattern);
-        Ok(Regex(fnmatch_regex::glob_to_regex(&escaped_str)?))
+        let regex_str = fnmatch_regex::glob_to_regex(&escaped_str)?.to_string();
+        Ok(Regex(fancy_regex::Regex::new(&regex_str).unwrap()))
+    }
+
+    pub fn from_regex_str(regex_str: &str) -> Result<Self, fancy_regex::Error> {
+        Ok(Regex(fancy_regex::Regex::new(
+            &remove_unsupported_escapes(regex_str),
+        )?))
     }
 
-    pub fn from_regex_str(regex_str: &str) -> Result<Self, regex::Error> {
-        Ok(Regex(regex::Regex::new(regex_str)?))
+    pub fn is_match(&self, string: &str) -> bool {
+        self.0.is_match(string).expect("Match failed")
```

Comment:
```suggestion
        self.0.is_match(string).unwrap_or_default()
```
Default to `false` but we may want to return the error instead :thinking:

Reply:
We'll drop `fancy_regex` so this result will no longer be there